File: back/src/main/java/br/eng/eliseu/presente/model/filter/TamanhoFilter.java
package br.eng.eliseu.presente.model.filter;

import br.eng.eliseu.presente.model.ProdutoTipoEnum;
import br.eng.eliseu.presente.model.core.BaseFilter;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class TamanhoFilter extends BaseFilter {

    private ProdutoTipoEnum tipo;

    private String tamanho;

}

File: front/src/app/shared/model/filter/tamanho-filter.ts
import {BaseFilter} from "@/shared/model/core/base-filter";
import {ProdutoTipoEnum} from "@/shared/model/enum/produto-tipo.enum";

export class TamanhoFilter extends BaseFilter {

    tipo!: ProdutoTipoEnum;
    tamanho!: string;

    constructor(init?: Partial<TamanhoFilter>) {
        super();
        Object.assign(this, init);
    }
}
